### PR TITLE
Upgrade nebula.netflixoss to replace bintray publication and replace TravisCI with Github Actions

### PR DESCRIPTION
**Pull request by rpalcolea**
Hi folks,

As you might be aware, JFrog is sunsetting Bintray and JCenter: https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter/

This upgrades to Nebula NetflixOSS which will publish to Nebula's OSS repos and Maven Central.

Repositories based on status:

* https://netflixoss.jfrog.io/artifactory/maven-oss-snapshots
* https://netflixoss.jfrog.io/artifactory/maven-oss-candidates
* https://netflixoss.jfrog.io/artifactory/maven-oss-releases

Few notes in other changes:

* Upgraded to latest Gradle
* Added lombok to `annotationProcessor` configuration 
* Upgraded shadow plugin
*

### CODE REVIEW

**Comment by jeffchao** [comment]
Looking at this today.

**Comment by rpalcolea** [comment]
Thanks @jeffchao , I'll try to address other repos through the week to reduce the migration burden on you folks as it requires gradle upgrades and such in most of cases.

We are also exploring the idea of introducing a Github action to keep folks in latest nebula.netflixoss plugin + gradle wrapper. More details to come once we have clarity on this

**Comment by jeffchao** [comment]
@rpalcolea BTW we've moved some of our repos into this one. The ones that were moved are marked `DEPRECATED` in their READMEs, so we don't necessarily have to migrate those (we can just delete/archive those repos some time).

Here's the list of repos outside of this one which are _still active_:

1. https://github.com/Netflix/mantis-api
2. https://github.com/Netflix/mantis-mql
3. https://github.com/Netflix/mantis-rxcontrol
4. https://github.com/Netflix/mantis-gradle-plugin
5. https://github.com/Netflix/mantis-rxnetty

**Comment by rpalcolea** [comment]
thanks for the heads up @jeffchao 

I found a problem with travisci encrypted files + forks... I also added a change to move you to Github Actions and follow the same pattern that you had in Travis for CI/releasing

**Comment by jeffchao** [comment]
Awesome thanks @rpalcolea . Yeah we've been looking at moving off of Travis when we got the first notice awhile back. I'll look at your changes.

**Comment by rpalcolea** [comment]
Hi @jeffchao , I went ahead and opened PRs for the repos you mentioned above, including the move to GHA.

**Comment by rpalcolea** [comment]
Hi @jeffchao , just a quick reminder on this :) and the other PRs:


https://github.com/Netflix/mantis-api/pull/20
https://github.com/Netflix/mantis-rxcontrol/pull/9
https://github.com/Netflix/mantis-rxnetty/pull/4
https://github.com/Netflix/mantis-gradle-plugin/pull/2
https://github.com/Netflix/mantis-mql/pull/3

**Comment by jeffchao** [comment]
> Hi @jeffchao , just a quick reminder on this :) and the other PRs:
> 
> [Netflix/mantis-api#20](https://github.com/Netflix/mantis-api/pull/20)
> [Netflix/mantis-rxcontrol#9](https://github.com/Netflix/mantis-rxcontrol/pull/9)
> [Netflix/mantis-rxnetty#4](https://github.com/Netflix/mantis-rxnetty/pull/4)
> [Netflix/mantis-gradle-plugin#2](https://github.com/Netflix/mantis-gradle-plugin/pull/2)
> [Netflix/mantis-mql#3](https://github.com/Netflix/mantis-mql/pull/3)

Yep thanks. I was going to circle back and get these in tomorrow.

File: mantis-server/mantis-server-worker/build.gradle [hunk 1]
```diff
@@ -41,6 +41,7 @@ dependencies {
     }
     compile "org.apache.httpcomponents:httpclient:$httpComponentsVersion"
     compile "io.mantisrx:mantis-rxcontrol:$mantisRxControlVersion"
+    compile "com.yahoo.datasketches:sketches-core:0.9.1"
```
Comment (jeffchao) [inline mantis-server/mantis-server-worker/build.gradle:44]: @rpalcolea what's the motivation for this addition?

File: mantis-examples/mantis-examples-mantis-publish-sample/build.gradle [hunk 2]
```diff
@@ -34,7 +34,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.github.jengelman.gradle.plugins:shadow:4.0.4'
+        classpath 'com.github.jengelman.gradle.plugins:shadow:6.1.0'
```
Comment (jeffchao) [inline mantis-examples/mantis-examples-mantis-publish-sample/build.gradle:37]: @rpalcolea any breaking changes between these versions which would prevent shading? Asking to see if we need to change any other gradle configs.

File: mantis-shaded/build.gradle [hunk 3]
```diff
@@ -52,5 +52,5 @@ shadowJar {
     mergeServiceFiles()
 }
 
-jar.deleteAllActions()
+jar.enabled = false
```
Comment (jeffchao) [inline mantis-shaded/build.gradle:55]: @rpalcolea what's the effect of this change?

Comment (rpalcolea) [reply mantis-shaded/build.gradle:55]: `deleteAllActions` was meant to remove all actions in a task, similar to `enabled`

`deleteAllActions` was deprecated and removed in newer versions of Gradle so enabled is the natural replacement here

File: mantis-server/mantis-server-worker/build.gradle (hunk 1 above)
Comment (rpalcolea) [reply mantis-server/mantis-server-worker/build.gradle:44]: You were using this library in you code transitively, however this library came transitively:

```
com.yahoo.datasketches:sketches-core:0.9.1
+--- runtimeClasspath
\--- io.mantisrx:mantis-rxcontrol:1.3.17
     \--- runtimeClasspath (requested io.mantisrx:mantis-rxcontrol:1.3.+)
```

If you look at https://search.maven.org/artifact/io.mantisrx/mantis-rxcontrol/1.3.17/jar 

```
    <dependency>
      <groupId>com.yahoo.datasketches</groupId>
      <artifactId>sketches-core</artifactId>
      <version>0.9.1</version>
      <scope>runtime</scope>
    </dependency>
```

It comes in `runtime` scope

  In Gradle 5.x, the compile/runtime classpath separation was introduced which means that runtime dependencies won't be available in compile classpath. This is why I added it.

File: mantis-examples/mantis-examples-mantis-publish-sample/build.gradle (hunk 2 above)
Comment (rpalcolea) [reply mantis-examples/mantis-examples-mantis-publish-sample/build.gradle:37]: I don't think so. It is more about compatibility with modern versions of Gradle other than breaking changes. We upgraded multiple projects/apps without issues

The main breaking changes in shadow plugin came from 2.x to 4.x but not your case

File: mantis-server/mantis-server-worker/build.gradle (hunk 1 above)
Comment (jeffchao) [reply mantis-server/mantis-server-worker/build.gradle:44]: Got it. Thanks for catching that.

**Comment by jeffchao** [comment]
Thanks @rpalcolea !

**Pull request merged**